pyomo/contrib/sensitivity_toolbox/sens.py

```diff
@@ -218,7 +218,501 @@ def sipopt(instance, paramSubList, perturbList,
+                b.constList.add(expr=ExpressionReplacementVisitor(
+                    substitute=variableSubMap,
+                    remove_named_expressions=True).dfs_postorder_stack(
+                        cc.lower) <= ExpressionReplacementVisitor(
+                            substitute=variableSubMap,
+                            remove_named_expressions=
+                            True).dfs_postorder_stack(cc.body)
+                    )
+                # Add constraint for upper bound
+                b.constList.add(expr=ExpressionReplacementVisitor(
+                    substitute=variableSubMap,
+                    remove_named_expressions=True).dfs_postorder_stack(
+                        cc.upper) >= ExpressionReplacementVisitor(
+                            substitute=variableSubMap,
+                            remove_named_expressions=
+                            True).dfs_postorder_stack(cc.body)
+                    )
+        cc.deactivate()
+
+    # paramData to varData constraint list
+    b.paramConst = ConstraintList()
+    for ii in paramDataList:
+        jj=variableSubMap[id(ii)]
+        b.paramConst.add(ii==jj)
+        
+    # Declare Suffixes
+    m.sens_state_0 = Suffix(direction=Suffix.EXPORT)
+    m.sens_state_1 = Suffix(direction=Suffix.EXPORT)
+    m.sens_state_value_1 = Suffix(direction=Suffix.EXPORT)
+    m.sens_init_constr = Suffix(direction=Suffix.EXPORT)
+    
+    m.sens_sol_state_1 = Suffix(direction=Suffix.IMPORT)
+    m.sens_sol_state_1_z_L = Suffix(direction=Suffix.IMPORT)
+    m.sens_sol_state_1_z_U = Suffix(direction=Suffix.IMPORT)
+    
+    
+    # for reasons that are not entirely clear, 
+    #     ipopt_sens requires the indices to start at 1
+    kk=1
+    for ii in paramDataList:
+        m.sens_state_0[variableSubMap[id(ii)]] = kk
+        m.sens_state_1[variableSubMap[id(ii)]] = kk
+        m.sens_state_value_1[variableSubMap[id(ii)]] = \
+                                                   value(perturbSubMap[id(ii)])
+        m.sens_init_constr[b.paramConst[kk]] = kk
+        kk += 1    
+    
+    if method == 'sipopt':
+        # Send the model to the ipopt_sens and collect the solution
+        results = opt.solve(m, keepfiles=keepfiles, tee=streamSoln)
+    elif method == 'kaug':
+        m.dual = Suffix(direction=Suffix.IMPORT_EXPORT)
+        m.ipopt_zL_out = Suffix(direction=Suffix.IMPORT)
+        m.ipopt_zU_out = Suffix(direction=Suffix.IMPORT)
+        m.ipopt_zL_in = Suffix(direction=Suffix.EXPORT)
+        m.ipopt_zU_in = Suffix(direction=Suffix.EXPORT)    
+        m.dcdp = Suffix(direction=Suffix.EXPORT)  #: the constraint "order"  (integer >0)
+        m.DeltaP = Suffix(direction=Suffix.EXPORT)  #: the parameter values difference (float)
+        kk = 1
+        for ii in paramDataList:
+            m.dcdp[b.paramConst[kk]] = kk
+            m.DeltaP[b.paramConst[kk]] = value(ii)-value(perturbSubMap[id(ii)])
+            kk += 1
+        print('ipopt \n\n\n')
```

Comment:
@jsiirola @blnicho Dan suggested using a logger instead of print. Does Pyomo have any preference?

Reply:
Yes.  Please use a logger obtained from the module namespace (e.g., `logger = logging.getLogger(__name__)`).

Reply:
I changed them with logger. Thanks,